be/src/runtime/exec_env_init.cpp

```diff
@@ -121,7 +122,10 @@ Status ExecEnv::_init(const std::vector<StorePath>& store_paths) {
     _init_mem_tracker();
 
     RETURN_IF_ERROR(_load_channel_mgr->init(_mem_tracker->limit()));
-
+    _heartbeat_flags = new HeartbeatFlags();
+    if (_storage_engine != nullptr) {
```

Comment:
Prefer doing it outside of ExecEnv. ExecEnv is just a singleton container, it should not be aware of any dependence of its elements.

Reply:
done, move it to doris_main